coops-ui/src/components/AnnualReport/Certify.vue

```diff
@@ -57,38 +57,39 @@ export default class Certify extends Vue {
    * Lifecycle callback to always give the parent a "valid" event for its property values.
    */
   private created (): void {
-    this.emitValid(this.trimmedCertifiedBy && this.isCertified)
+    this.emitValid(Boolean(this.trimmedCertifiedBy && this.isCertified))
   }
 
   /**
    * Computed value.
    * @returns The trimmed "Certified By" string (may be '').
    */
   private get trimmedCertifiedBy (): string {
-    return this.certifiedBy && this.certifiedBy.trim()
+    // remove repeated inline whitespace, and leading/trailing whitespace
+    return this.certifiedBy && this.certifiedBy.replace(/\s+/g, ' ').trim()
   }
 
-  // Emit an update event.
+  // Emit an update event when input changes.
   @Emit('update:certifiedBy')
-  private emitCertifiedBy (val: string): string {
-    this.emitValid(val && val.trim() && this.isCertified)
+  private emitCertifiedBy (certifiedBy: string): string {
+    // remove repeated inline whitespace, and leading/trailing whitespace
+    certifiedBy = certifiedBy && certifiedBy.replace(/\s+/g, ' ').trim()
+    this.emitValid(Boolean(certifiedBy && this.isCertified))
 
-    return val
+    return certifiedBy
   }
 
-  // Emit an update event.
+  // Emit an update event when checkbox changes.
   @Emit('update:isCertified')
-  private emitIsCertified (val: boolean): boolean {
-    this.emitValid(this.trimmedCertifiedBy && !!val)
+  private emitIsCertified (isCertified: boolean): boolean {
+    this.emitValid(Boolean(this.trimmedCertifiedBy && isCertified))
 
-    return val
+    return isCertified
```

Comment:
Is this method just returning the parameter passed in?

Reply:
Yes and yes.

There are 2 ways to implement an emitter method in TS. I used this way because the body needs to emit the valid flag as well as emit the parameter passed in. An example of the other way is on line 92, below.